### PR TITLE
[ARM] `managedapp definition update`: Add new command for updating managedapp definition

### PR DESCRIPTION
**Pull request by zhoxing-ms**
Feature request: #16740

**Description**
In the past, `managedapp definition create` could support both adding and updating, it is not a good design to update with the command named create.
So in order to make the naming of the command consistent with the actual function, a new command with the same function as `managedapp definition create` is added.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
ARM

**Review by evelyn-ys — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/resource/_params.py [hunk 1]
```diff
@@ -508,11 +508,12 @@ def load_arguments(self, _):
...
-        c.argument('authorizations', options_list=['--authorizations', '-a'], nargs='+', help="space-separated authorization pairs in a format of `<principalId>:<roleDefinitionId>`")
-        c.argument('createUiDefinition', options_list=['--create-ui-definition', '-c'], help='JSON formatted string or a path to a file with such content', type=file_type)
-        c.argument('mainTemplate', options_list=['--main-template', '-t'], help='JSON formatted string or a path to a file with such content', type=file_type)
+    for operation in ['create', 'update']:
+        with self.argument_context('managedapp definition {}'.format(operation)) as c:
+            c.argument('lock_level', arg_type=get_enum_type(ApplicationLockLevel), help='The type of lock restriction.')
+            c.argument('authorizations', options_list=['--authorizations', '-a'], nargs='+', help="space-separated authorization pairs in a format of `<principalId>:<roleDefinitionId>`")
+            c.argument('createUiDefinition', options_list=['--create-ui-definition', '-c'], help='JSON formatted string or a path to a file with such content', type=file_type)
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/resource/_params.py:515]: `createUiDefinition` is not pythonic and the variable in custom function is `create_ui_definition`. Please take a look.

File: src/azure-cli/azure/cli/command_modules/resource/_params.py [hunk 2]
```diff
@@ -508,11 +508,12 @@ def load_arguments(self, _):
...
+            c.argument('mainTemplate', options_list=['--main-template', '-t'], help='JSON formatted string or a path to a file with such content', type=file_type)
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/resource/_params.py:516]: the same

File: src/azure-cli/azure/cli/command_modules/resource/_params.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/resource/_params.py]: Good catch! I have modified it

Comment (yonzhan) [reply src/azure-cli/azure/cli/command_modules/resource/_params.py]: You are Java expert:)

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/resource/_params.py]: @yonzhan Actually, this code is not written by me, but legacy code 😂

File: src/azure-cli/azure/cli/command_modules/resource/_params.py [hunk 3]
```diff
@@ -508,11 +508,12 @@ def load_arguments(self, _):
...
-        c.argument('authorizations', options_list=['--authorizations', '-a'], nargs='+', help="space-separated authorization pairs in a format of `<principalId>:<roleDefinitionId>`")
-        c.argument('createUiDefinition', options_list=['--create-ui-definition', '-c'], help='JSON formatted string or a path to a file with such content', type=file_type)
-        c.argument('mainTemplate', options_list=['--main-template', '-t'], help='JSON formatted string or a path to a file with such content', type=file_type)
+    for operation in ['create', 'update']:
+        with self.argument_context('managedapp definition {}'.format(operation)) as c:
+            c.argument('lock_level', arg_type=get_enum_type(ApplicationLockLevel), help='The type of lock restriction.')
+            c.argument('authorizations', options_list=['--authorizations', '-a'], nargs='+', help="space-separated authorization pairs in a format of `<principalId>:<roleDefinitionId>`")
+            c.argument('create_ui_definition', options_list=['--create-ui-definition', '-c'], help='JSON formatted string or a path to a file with such content', type=file_type)
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/resource/_params.py:515]: Could you check the parameter only support file path or JSON format could also work as it is marked as `file_type`?

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/resource/_params.py:515]: Yes, it supports JSON format string and file path.
For file path, there is a test to ensure support [code link](https://github.com/Azure/azure-cli/blob/11e92414c4f5539539101d1efcbf984d8ba503d7/src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py#L2815
)

**Review by Juliehzl — APPROVED** [review]

**Pull request merged**